### ISSUE TITLE
[Bug]: Library not updated after uploading from the upload page

### ISSUE THREAD

**Issue by kymahi** [advplyr/audiobookshelf#1875]
### Describe the issue

Myself and another user have recently uploaded books to our server, but the books are not showing up in the library until I kick off a manual scan of the library.

### Steps to reproduce the issue

1. Go to server
2. click Upload icon
3. upload audiobook (I filled out author and series)
4. go to library
5. find no new book

### Audiobookshelf version

2.2.23

### How are you running audiobookshelf?

Docker

**Comment by advplyr** [comment]
Do you have the Watcher disabled?

**Comment by kymahi** [comment]
No it's not disabled :\

**Comment by advplyr** [comment]
If you can upload then share the logs with debug logs turned on that can be helpful to see why the watcher is not picking it up.

**Comment by AquaMorph** [comment]
I also ran into this and I deleted the book file and tried to collect a log this time but it fails to upload now full out.

**Comment by AquaMorph** [comment]
I think the issue of not being able to upload was unrelated. I did a full reboot of my entire tech stack and I was able to upload and the error reproduced. Here is the log:
```
2023-07-12 01:20:26

INFO

Uploading 1 files to /audiobooks/WeAreSatellites_ep7

2023-07-12 01:20:28

DEBUG

[Watcher] File Added /audiobooks/WeAreSatellites_ep7/WeAreSatellites_ep7.m4b

2023-07-12 01:20:28

DEBUG

[Watcher] Modified file in library "Christian's Audiobooks" and folder "fol_0axqmpomam0yt18q4y" with relPath "/WeAreSatellites_ep7/WeAreSatellites_ep7.m4b"

2023-07-12 01:20:32

INFO

[Server] 1 Files Changed

2023-07-12 01:20:32

DEBUG

[Scanner] Scanning file update groups in folder "fol_0axqmpomam0yt18q4y" of library "Christian's Audiobooks"

2023-07-12 01:20:32

DEBUG

[Scanner] scanFolderUpdates fileUpdateGroup [object Object]

2023-07-12 01:20:32

DEBUG

[Scanner] Folder update group must be a new item "WeAreSatellites_ep7" in library "Christian's Audiobooks"

2023-07-12 01:20:32

DEBUG

[Scanner] Scanning new item "/audiobooks/WeAreSatellites_ep7"

2023-07-12 01:20:32

ERROR

[MediaFileScanner] Invalid data found when processing input : "/audiobooks/WeAreSatellites_ep7/WeAreSatellites_ep7.m4b"

2023-07-12 01:20:32

WARN

[Scanner] Library item has no media files "/audiobooks/WeAreSatellites_ep7"

2023-07-12 01:20:32

DEBUG

[Scanner] Folder scan results [object Object]

2023-07-12 01:20:35

DEBUG

Setting permission "484" for uid 99 and gid 100 | "/audiobooks"
```

**Comment by AquaMorph** [comment]
Going to libraries and selecting the library drop down and running a scan caused it to show up as expected.

**Comment by advplyr** [comment]
Ah, I understand what happened. This is the same issue as #1362 and we can merge this issue into that one.

tldr; the watcher is detecting the file before it has finished uploading so when the scanner fails to scan it. For the most part this is handled well by the watcher library but I think slow file transfer can cause this.

**Comment by advplyr** [comment]
Duplicate of #1362
